apps/ledger-live-mobile/src/screens/Protect/RedirectToRecoverStaxFlow.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from "react";
+import { useNavigation } from "@react-navigation/native";
+import { DeviceModelId } from "@ledgerhq/types-devices";
+import { BaseNavigatorStackParamList } from "../../components/RootNavigator/types/BaseNavigator";
+import {
+  RootComposite,
+  StackNavigatorProps,
+} from "../../components/RootNavigator/types/helpers";
+import { NavigatorName, ScreenName } from "../../const";
+import { useNavigationInterceptor } from "../Onboarding/onboardingContext";
+
+type NavigationProps = RootComposite<
+  StackNavigatorProps<
+    BaseNavigatorStackParamList,
+    ScreenName.RedirectToOnboardingRecoverFlow
+  >
+>;
+
+export function RedirectToRecoverStaxFlowScreen() {
+  const { replace } = useNavigation<NavigationProps["navigation"]>();
+  const { setShowWelcome, setFirstTimeOnboarding } = useNavigationInterceptor();
+
+  useEffect(() => {
+    setShowWelcome(false);
+    setFirstTimeOnboarding(false);
+    replace(NavigatorName.Base, {
+      screen: ScreenName.BleDevicePairingFlow,
+      params: {
+        filterByDeviceModelId: DeviceModelId.stax,
+        areKnownDevicesDisplayed: true,
+        onSuccessAddToKnownDevices: false,
+        isRecoverFlow: true,
+        // `onSuccessNavigateToConfig` will never be used when `isRecoverFlow: true`
+        onSuccessNavigateToConfig: {
+          navigationType: "push",
+          navigateInput: {
+            name: NavigatorName.BaseOnboarding,
+            params: {
+              screen: NavigatorName.SyncOnboarding,
+              params: {
+                screen: ScreenName.SyncOnboardingCompanion,
+                params: {
+                  device: null,
+                },
+              },
+            },
+          },
+          pathToDeviceParam: "params.params.params.device",
+        },
+      },
+    });
+  }, [replace, setFirstTimeOnboarding, setShowWelcome]);
+
+  return <></>;
```

Comment:
For this use case you should use directly the component `apps/ledger-live-mobile/src/components/BleDevicePairingFlow/index.tsx` in here rather than redirecting to the pairing flow screen and having to implement a logic specific to your use case in that screen. It will be much simpler